src/psyclone/tests/dependency_test.py

```diff
@@ -0,0 +1,329 @@
+
+
+def test_if_statement(parser):
+    ''' Tests handling an if statement
+    '''
+    reader = FortranStringReader('''program test_prog
+                                 if (a .eq. b) then
+                                    p(i) = q(i)
+                                 else
+                                   q(i) = r(i)
+                                 endif
+                                 end program test_prog''')
+    ast = parser(reader)
+    psy = PSyFactory(API).create(ast)
+    schedule = psy.invokes.get("test_prog").schedule
+
+    if_stmt = schedule.children[0]
+    assert isinstance(if_stmt, IfBlock)
+    var_accesses = VariablesAccessInfo()
+    if_stmt.reference_accesses(var_accesses)
+    assert str(var_accesses) == "a: READ, b: READ, i: READ, p: WRITE, "\
+                                "q: READWRITE, r: READ"
+    # Test that the two accesses to 'q' indeed show up as
+    q_accesses = var_accesses.get_varinfo("q").get_all_accesses()
+    assert len(q_accesses) == 2
+    assert q_accesses[0].get_access_type() == AccessType.READ
+    assert q_accesses[1].get_access_type() == AccessType.WRITE
+    assert q_accesses[0].get_location() < q_accesses[1].get_location()
+
+
+@pytest.mark.xfail(reason="Calls in nemo not yet supported")
+def test_call(parser):
+    ''' Check that we correctly handle a call in a program '''
+    reader = FortranStringReader('''program test_prog
+                                 call sub(a,b)
+                                 end program test_prog''')
+    ast = parser(reader)
+    psy = PSyFactory(API).create(ast)
+    schedule = psy.invokes.get("test_prog").schedule
+
+    code_block = schedule.children[0]
+    call_stmt = code_block.statements[0]
+    var_accesses = VariablesAccessInfo()
+    call_stmt.reference_accesses(var_accesses)
+    assert str(var_accesses) == "a: UNKNOWN, b: UNKNOWN"
+
+
+def test_do_loop(parser):
+    ''' Check the handling of do loops.
+    TODO: this only tests a nemo do loop. At this stage the loop
+    boundaries in nemo are only strings (not instances of Reference or so),
+    so in case of a loop like: "do jj=1, n", N would
+    '''
+    reader = FortranStringReader('''program test_prog
+                                 do jj=1, 10
+                                    do ji=1, 10
+                                       s(ji, jj)=t(ji, jj)+1
+                                    enddo
+                                 enddo
+                                 end program test_prog''')
+    ast = parser(reader)
+    psy = PSyFactory(API).create(ast)
+    schedule = psy.invokes.get("test_prog").schedule
+
+    do_loop = schedule.children[0]
+    assert isinstance(do_loop, nemo.NemoLoop)
+    var_accesses = VariablesAccessInfo()
+    do_loop.reference_accesses(var_accesses)
+    assert str(var_accesses) == "ji: READWRITE, jj: READWRITE, s: WRITE, "\
+                                "t: READ"
+
+
+@pytest.mark.xfail(reason="Nemo converts all loop limits to strings")
```

Comment:
Issue number please (might be #400) and strictly it should be "NEMO API converts..."